shared/fs/nav-header/mobile-header.js

```diff
@@ -15,26 +16,44 @@ type Props = {|
   onBack: () => void,
   path: Types.Path,
 |}
+type State = {|
+  filterExpanded: boolean,
+|}
 
-const MobileHeader = (props: Props) => (
-  <Kb.Box2
-    direction="vertical"
-    fullWidth={true}
-    style={Styles.collapseStyles([styles.container, props.bannerType === 'offline' && styles.blue])}
-  >
-    {props.bannerType === 'offline' && <Kb.Banner text="You are offline." color="blue" />}
-    <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.expandedTopContainer}>
-      <Kb.BackButton badgeNumber={0 /* TODO */} onClick={props.onBack} />
-      <Kb.Box style={styles.gap} />
-      <Actions path={props.path} />
-    </Kb.Box2>
-    <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.expandedTitleContainer}>
-      <Kb.Text type="Header" lineClamp={1}>
-        {props.path === Constants.defaultPath ? 'Files' : Types.getPathName(props.path)}
-      </Kb.Text>
-    </Kb.Box2>
-  </Kb.Box2>
-)
+class MobileHeader extends React.PureComponent<Props, State> {
+  state = {filterExpanded: false}
+  _triggerFilterMobile = () => {
+    this.setState({filterExpanded: true})
+  }
+  _filterDone = () => {
+    this.setState({filterExpanded: false})
+  }
+  render() {
+    return (
+      <Kb.Box2
+        direction="vertical"
+        fullWidth={true}
+        style={Styles.collapseStyles([styles.container, this.props.bannerType === 'offline' && styles.blue])}
+      >
+        {this.props.bannerType === 'offline' && <Kb.Banner text="You are offline." color="blue" />}
+        {this.state.filterExpanded ? (
+          <Kbfs.FolderViewFilter path={this.props.path} onBlur={this._filterDone} />
+        ) : (
+          <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.expandedTopContainer}>
+            <Kb.BackButton badgeNumber={0 /* TODO */} onClick={this.props.onBack} />
```

Comment:
What's the badge supposed to contain? do you plan to do it in this PR?

Reply:
This is actually from previous commit so not planning to address it in this PR. Not sure what it should be yet. Perhaps notifications from previous screen?

Reply:
Do back buttons normally have badges on them? I feel like the TODO in the code is useless without any sort of explanation, so if nobody knows what it was even for we should just delete the `badgeNumber` from the `BackButton`.

Reply:
It's in design actually: https://zpl.io/amygjBK , so perhaps we'll still need it in the future. I'll make a ticket for it.

Reply:
Can you stick the ticket number in the TODO?